packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -201,6 +201,29 @@ export default class CashierStore extends BaseStore {
         this.onSwitchAccount(this.accountSwitcherListener);
     }
 
+    // Initial init just for displaying P2P banner without mounting the entire cashier
+    @action.bound
+    init() {
+        // eslint-disable-next-line no-undef
+        reaction(
+            () => [this.root_store.client.is_logged_in, this.root_store.client.residence],
+            () => {
+                if (!this.is_p2p_visible && !this.root_store.client.is_virtual) {
+                    WS.authorized.p2pAdvertiserInfo().then(advertiser_info => {
+                        const advertiser_error = getPropertyValue(advertiser_info, ['error', 'code']);
+                        if (advertiser_error === 'RestrictedCountry') {
+                            this.setIsP2PBannerVisible(false);
+                        } else {
+                            this.setIsP2PBannerVisible(true);
+                        }
+
+                        this.is_p2p_advertiser = !advertiser_error;
```

Comment:
do we need to set this in init?

Reply:
(if keeping two separate instances of calling advertiser info, doesn't make sense for this to be here. if we are refactoring to keep one instance of it, then it makes sense to have this here)

Reply:
with the new changes, we can remove is_p2p_advertiser, i dont see any usage at all for this now @sean-binary